### DISCUSSION TITLE
About the formatted messages in chatbox

### DISCUSSION THREAD

**Discussion by jebjordan** [IntelligenceModding/AdvancedPeripherals#207] [Q&A] [answered]
How do I actually use this?
i tried just using stuff like {"color":"red", "text":"a"} and other such things, since i assumed it used the format from [here](https://minecraft.fandom.com/wiki/Raw_JSON_text_format)

Whenever I use this command though, it simply chats the string I put into the command rather than actually outputting formatted.
Very confused.

@Seniorendi (I don't know if this will ping you or not... hm.
Anyways, do you know what's wrong here? Is this a bug or am I just not understanding how to use this function.

**Comment by SirEndii** [comment]
You can use color codes like §b for aqua

Sorry for the late answer

**Comment by 0xadg** [comment, marked as answer]
I'm not very familiar with Java, but there's a questionable code in ChatBoxPeripheral.java
```
public final MethodResult sendFormattedMessage(@Nonnull IArguments arguments) throws LuaException {
        return withChatOperation(ignored -> {
            String message = arguments.getString(0);
            IFormattableTextComponent component = ITextComponent.Serializer.fromJson(message);
            if (component == null)
                return MethodResult.of(null, "incorrect json");
            IFormattableTextComponent preparedMessage = new StringTextComponent(String.format(PREFIX_FORMAT, arguments.optString(1, AdvancedPeripheralsConfig.defaultChatBoxPrefix))).append(message);
            for (ServerPlayerEntity player : ServerLifecycleHooks.getCurrentServer().getPlayerList().getPlayers()) {
                player.sendMessage(preparedMessage, Util.NIL_UUID);
            }
            return MethodResult.of(true);
        });
    }
```
`IFormattableTextComponent component`, which is probably formatted JSON text, is used only in null check and never used anywhere further, and `preparedMessage` is created from prefix and `message`, which is just input string. Maybe that's an origin of this problem?

**Reply by SirEndii** [reply]
Uh this is true
@SirEdvin

**Reply by SirEdvin** [reply]
Okey, so, first, thanks for investigation, this is very stupid mistake. 

I will fix this right now, it will work correct (tested already)

**Reply by jebjordan** [reply]
lol.
Well glad I asked the question then, haha.
